certbot-apache/certbot_apache/configurator.py

```diff
@@ -1012,13 +1012,34 @@ def _add_servername_alias(self, target_name, vhost):
             self.parser.find_dir("ServerAlias", target_name,
                                  start=vh_path, exclude=False)):
             return
+        if self._has_matching_wildcard(vh_path, target_name):
+            return
         if not self.parser.find_dir("ServerName", None,
                                     start=vh_path, exclude=False):
             self.parser.add_dir(vh_path, "ServerName", target_name)
         else:
             self.parser.add_dir(vh_path, "ServerAlias", target_name)
         self._add_servernames(vhost)
 
+    def _has_matching_wildcard(self, vh_path, target_name):
```

Comment:
Apache2 supports both '*' and '?' wildcards, to claim that we support them also, we'll need to match '?' wildcards too.